app/models/coaching.rb

```diff
@@ -0,0 +1,10 @@
+class Coaching < ApplicationRecord
+  self.inheritance_column = :sports_type
+
+  has_many :batches, dependent: :restrict_with_error
+  has_many :registrations, through: :batches
+  has_many :players, through: :registrations
+
+  validates :head_name, :sports_type, presence:  true
```

Comment:
spacing

Reply:
Is it mandatory to add presence validation to STI column? what happens if we don't?

Reply:
Spacing Fixed.
If we don't add this validation and enters a nil value in type column, it will be saved, but can not be accessible in any inherited models.